msibi/pair.py

```diff
@@ -47,32 +48,59 @@ def __init__(self, type1, type2, potential, head_correction_form="linear"):
         self.head_correction_form = head_correction_form
 
     def add_state(
-        self, state, target_rdf, alpha, pair_indices=None, alpha_form="linear"
+        self,
+        state,
+        target_rdf=None,
+        calculate_target_rdf=False,
+        pair_indices=None,
+        alpha_form="linear"
     ):
         """Add a state to be used in optimizing this pair.
 
         Parameters
         ----------
         state : State
             A state object.
-        target_rdf : np.ndarray, shape=(n_bins, 2), dtype=float
-            Coarse-grained target RDF.
-        alpha : float
-            The alpha value used to scale the weight of this state.
         pair_indices : array-like (n_pairs, 2) dtype=int
             Each row gives the indices of two atoms representing a pair
             (default None)
         alpha_form : str
             For alpha as a function of r, gives form of alpha function
             (default 'linear')
         """
+        if calculate_target_rdf and target_rdf != None:
+            raise ValueError(
+                    "Setting calcualte_target_rdf = True will overwirte "
+                    "the data passed to target_rdf. calculate_target_rdf "
+                    "should only be used when target_rdf is None"
+                    )
+        if target_rdf:
+            if os.path.isfile(target_rdf):
+                try:
+                    target_rdf = np.loadtxt(target_rdf)
+                except Exception as e:
+                    print(e)
+            elif isinstance(target_rdf, np.ndarray):
+                pass
+
+        elif calculate_target_rdf:
+            target_rdf = state_pair_rdf(state, self)
+
+        if len(target_rdf) != state.opt.n_rdf_points:
+            raise ValueError(
+                    "The target RDF passed is not the same length as "
+                    "n_rdf_points set during the initialization of the "
+                    "MSIBI() class."
+                    )
```

Comment:
What would be set automatically, `state.opt.n_rdf_points`? That comes from the `MSIBI` class which is created before State objects are created, so I don't see how it could be set automatically.  

This is kind of related to the second comment I left in #10.  If the user is going to create target RDFs separately, then how do we enforce that they were created with the same parameters that the query RDFs will be created? Like, same cutoff, number of x-axis bins, etc...

The parameters defined in the `MSIBI` class are automatically used when `calculate_target_rdf = True` and when calculating the query RDF.  So, in that sense this is handled automatically and the error wouldn't come up.  If the user is passing already generated RDFs, then they have to make sure the rdf related parameters passed to `MSIBI` match what was used when creating the target RDFs.

Reply:
I guess, do we want the user to ever pass generated RDFs? If so, we could just use len(rdf) for that value. And maybe we can remove n_rdf_points as an argument to MSIBI.

Reply:
If the RDFs are computationally expensive to generate, then I think there's value in giving the user an option to pass in already generated data for a target_rdf.

But somewhere the number of rdf points has to be defined, before `calculate_target_rdf=True` is called.  That can either be in `MSIBI` or defined each time a state is created, but that introduces potential issues that would have to be checked as well (it's the same value for all).

The only other thing I can think of is to leave the `n_rdf_points` as an optional parameter in the `MSIBI` class, and if the user is passing in an already generated `target_rdf` then its length is used to set the `MSIBI.n_rdf_points` attribute which is then used later when finding the query RDF.

Reply:
But then that doesn't address the other issues that come up with mismatching RDF parameters.  This would make sure that the bin sizes are the same, but what about cutoff?  

Ultimately, I think it's ok to require the user to do some due diligence and understand the process enough to make sure they are passing in the correct info, like bin size and cutoff when initializing an optimization process.  Even if it means few extra checks are done. It's their responsibility to make sure that the same parameters they used when creating their own RDFs is passed into the `MSIBI` class.

Reply:
maybe a solution is to have both classes (state and pair) connected to the MSIBI object through some add_state/add_pair functions